### PR TITLE
adding console script to support Linux env

### LINKED ISSUE #328: Add console-script to start python under the embedding environment on linux
The embedding environment on linux is given by the .workbench_lite script.

We may want have a console-script installed with pymechanical to run a subcommand in this environment. It can be used to launch python or pytest, for example

### PR DESCRIPTION
**Pull request by dipinknair**
Add in .mechanical-env script which does the same functionality as .workbench_lite.

Fixes #328 & #401 

Related PR - #297

### CODE REVIEW

File: src/ansys/mechanical/core/run.py [hunk 1]
```diff
@@ -245,3 +245,121 @@ def cli(
...
+        exe, version = atp.find_mechanical(version=revision)
+    version = int(version * 10)
+
+    env = os.environ.copy()
+
+    # workbench (mechanical) installation directory
+    env["DS_INSTALL_DIR"] = os.path.dirname(exe)
+    env[f"AWP_ROOT{version}"] = f'{env.get("DS_INSTALL_DIR")}/..'
+
+    # Env vars used by workbench (mechanical) code
```
Comment (koubaa) [inline src/ansys/mechanical/core/run.py:292]: comare the .workbench_lite between 232 and 241, there might be some differences, for which you'll have to respect by using the `revision` argument

Comment (dipinknair) [reply src/ansys/mechanical/core/run.py:292]: Updated

File: src/ansys/mechanical/core/run.py [hunk 2]
```diff
@@ -245,3 +245,118 @@ def cli(
...
+
+    # Gets the revision number
+    if not revision:
+        exe, version = atp.find_mechanical()
+    else:
+        exe, version = atp.find_mechanical(version=revision)
+    version = int(version * 10)
+
+    env = os.environ.copy()
+
```
Comment (klmcadams) [inline src/ansys/mechanical/core/run.py:287]:
```suggestion
    AWP_ROOT = env.get(f"AWP_ROOT{version}")
    DS_INSTALL_DIR = env.get("DS_INSTALL_DIR")
```

Comment (klmcadams) [reply src/ansys/mechanical/core/run.py:287]: replace env.get(...) lines with AWP_ROOT and DS_INSTALL_DIR below to make it a little cleaner

**Review by klmcadams — COMMENTED** [review]
Good job! I just had some formatting suggestions you could implement if you would like

File: src/ansys/mechanical/core/run.py [hunk 3]
```diff
@@ -245,3 +245,118 @@ def cli(
...
+    Parameters
+    ----------
+    revision : int
+        The Ansys Revision number.
+
+    USAGE:
+    The following example demonstrates the main use of this tool:
+        $ mechanical-env -r 232 -- python test.py
+        $ mechanical-env -- make -C doc html
```
[the remark this answers is not in the document]
Comment (RobPasMue) [reply src/ansys/mechanical/core/run.py:272]: Has to be with "Examples" indeed

https://numpydoc.readthedocs.io/en/latest/format.html#examples

File: src/ansys/mechanical/core/run.py [hunk 4]
```diff
@@ -245,3 +245,107 @@ def cli(
...
+    env["MWRUNTIME"] = "1"
+    env["MWUSER_DIRECTORY"] = env.get("HOME") + "/.mw"
+    env["MWDONT_XCLOSEDISPLAY"] = "1"
+
+    # dynamic library preload
+    env["LD_PRELOAD"] = "libstdc++.so.6.0.28"
+
+    # dynamic library load path
+    env["LD_LIBRARY_PATH"] = (
+        f"{AWP_ROOT}/tp/stdc++:"
```
Comment (klmcadams) [inline src/ansys/mechanical/core/run.py:326]: The paths look good! :)

File: src/ansys/mechanical/core/run.py [hunk 5]
```diff
@@ -245,3 +245,107 @@ def cli(
...
+
+    env = os.environ.copy()
+    DS_INSTALL_DIR = os.path.dirname(exe)
+    AWP_ROOT = f"{DS_INSTALL_DIR}/.."
+
+    # workbench (mechanical) installation directory
+    env["DS_INSTALL_DIR"] = os.path.dirname(exe)
+    DS_INSTALL_DIR = env.get("DS_INSTALL_DIR")
+    env[f"AWP_ROOT{version}"] = f"{DS_INSTALL_DIR}/.."
+    AWP_ROOT = env.get(f"AWP_ROOT{version}")  # AWP_ROOT with revision number
```
[the remark this answers is not in the document]
Comment (dipinknair) [reply src/ansys/mechanical/core/run.py:298]: not needed, forgot to remove. Thanks for noticing that

[the remark this answers is not in the document]
Comment (klmcadams) [reply src/ansys/mechanical/core/run.py:298]: No problem!

**Review by koubaa — APPROVED** [review]

File: .mechanical-env [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+fi
+
+# Extract and print the captured values
+read -r version DS_INSTALL_DIR <<< "$output"
+
+# Envs
+awp_root="AWP_ROOT${version}"
+eval ${awp_root}=$DS_INSTALL_DIR/..
+
+# # Env vars used by workbench (mechanical) code
```
Comment (klmcadams) [inline .mechanical-env:56]:
```suggestion
# Env vars used by workbench (mechanical) code
```

**Review by klmcadams — APPROVED** [review]

File: doc/source/getting_started/running_mechanical.rst [hunk 7]
```diff
@@ -235,15 +235,15 @@ Additional information for Linux
 Starting with 2023 R2, it is possible to embed an instance of Mechanical on Linux.
 However, because of differences in how Mechanical works on Linux, you cannot simply
 run Python as usual. On Linux, certain environment variables must be set for the Python
-process before it starts. You can set up these environment variables using the ``.workbench_lite``
-script that is shipped with the Mechanical installation.
+process before it starts. You can set up these environment variables using the ``.mechanical-env``
+script that is with PyMechanical repository.
```
Comment (koubaa) [inline doc/source/getting_started/running_mechanical.rst:239]: it should be in the package, not in the repository

File: .github/workflows/ci_cd.yml [hunk 8]
```diff
@@ -306,7 +306,8 @@ jobs:
           ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL: 2
           NUM_CORES: 1
         run: |
-          xvfb-run /install/ansys_inc/v${{ needs.revn-variations.outputs.test_lic }}/aisol/.workbench_lite pytest -m embedding > pytest_output.txt || true
+          chmod +x .mechanical-env
```
Comment (koubaa) [inline .github/workflows/ci_cd.yml:309]: it should already have the execute permission

**Review by koubaa — CHANGES_REQUESTED** [review]

**Comment by RobPasMue** [comment]
Have a look at this @dipinknair - https://flit.pypa.io/en/latest/pyproject_toml.html#sdist-section

**Pull request closed without merging**